### PR TITLE
Add Datadog Exporter

### PR DESCRIPTION
**Pull request by dineshg13**
### What does this PR do?
The PR adds Datadog Exporter . It is thin wrapper around other exporter we have. 
Datadog Exporter in Agent and Datadog Exporter in collector will share the code for exporting each data type . 

We can't convert DD exporter as module at the moment as it has other agent dependencies. Will do so in the future PRs.

### CODE REVIEW

File: comp/otelcol/otlp/components/exporter/datadogexporter/config.go [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+)
+
+const (
+	// DefaultSite is the default site of the Datadog intake to send data to
+	DefaultSite = "datadoghq.com"
+)
+
+// APIConfig defines the API configuration options
+type APIConfig struct {
+	// Key is the Datadog API key to associate your Agent's data with your organization.
```
Comment (dineshg13) [inline comp/otelcol/otlp/components/exporter/datadogexporter/config.go:40]: This config is same as Datadog exporter config. In future we should ensure both exporters use same config.

File: comp/otelcol/otlp/components/exporter/datadogexporter/metadata.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// Type is the type of the exporter
+var Type = component.MustNewType("datadog")
+
+const (
+	// LogsStability is the stability level of the logs datatype.
+	LogsStability = component.StabilityLevelBeta
```
Comment (songy23) [inline comp/otelcol/otlp/components/exporter/datadogexporter/metadata.go:19]: This is alpha per https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/datadogexporter#datadog-exporter

**Review by songy23 — COMMENTED** [review]
One meta comment: a bunch of the configs do not apply to otel agent, e.g. `logs.dump_payloads`  `host_metadata` etc. If those are set should they result in config warnings?

File: comp/otelcol/otlp/components/exporter/datadogexporter/config.go [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+	// https://github.com/DataDog/datadog-agent/blob/505170c4ac8c3cbff1a61cf5f84b28d835c91058/pkg/trace/stats/concentrator.go#L55.
+	PeerTags []string `mapstructure:"peer_tags"`
+
+	// TraceBuffer specifies the number of Datadog Agent TracerPayloads to buffer before dropping.
+	// The default value is 0, meaning the Datadog Agent TracerPayloads are unbuffered.
+	TraceBuffer int `mapstructure:"trace_buffer"`
+
+	// flushInterval defines the interval in seconds at which the writer flushes traces
+	// to the intake; used in tests.
+	// flushInterval float64
```
[the remark this answers is not in the document]
Comment (dineshg13) [reply comp/otelcol/otlp/components/exporter/datadogexporter/config.go:112]: It is causing lint error, as we don't have trace agent yet.

File: comp/otelcol/otlp/components/exporter/datadogexporter/metadata.go (hunk 2 above)
Comment (dineshg13) [reply comp/otelcol/otlp/components/exporter/datadogexporter/metadata.go:19]: I think it should be beta in OTel Agent. in DD exporter we were waiting to use logs agent pipeline to mark it beta

**Comment by dineshg13** [comment]
@songy23 We should aim to have same config for both DD exporter in Agent vs DD exporter in collector. Hence I tried to replicate all the config. Some of the config would be noop for OTel agent

**Comment by songy23** [comment]
Yes - my point is we should warn the users in config_warnings for configs that are no-op in OTel agent. Otherwise they will wonder why the same config has effect in collector contrib but not in otel agent.

**Review by songy23 — APPROVED** [review]
Discussed offline, will make the change above in a follow-up PR.

File: comp/otelcol/otlp/components/exporter/datadogexporter/factory.go [hunk 4]
```diff
@@ -1,18 +1,15 @@
-// Unless explicitly stated otherwise all files in this repository are licensed
-// under the Apache License Version 2.0.
-// This product includes software developed at Datadog (https://www.datadoghq.com/).
-// Copyright 2021-present Datadog, Inc.
+// Copyright The OpenTelemetry Authors
```
Comment (songy23) [inline comp/otelcol/otlp/components/exporter/datadogexporter/factory.go:1]: License header seems wrong?

File: comp/otelcol/otlp/components/exporter/datadogexporter/factory.go [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+	_ component.Config,
+) (exporter.Logs, error) {
+	var logch chan *message.Message
+	if provider := f.logsAgent.GetPipelineProvider(); provider != nil {
+		logch = provider.NextPipelineChan()
+	}
+	lf := logsagentexporter.NewFactory(logch)
+	lc := &logsagentexporter.Config{
+		OtelSource:    "otel_agent",
+		LogSourceName: "otelcol",
```
Comment (songy23) [inline comp/otelcol/otlp/components/exporter/datadogexporter/factory.go:196]:
```suggestion
		LogSourceName: "otlp_log_ingestion",
```

as we discussed

Comment (liustanley) [inline comp/otelcol/otlp/components/exporter/datadogexporter/factory.go:196]: This should be set to `logsagentexporter.logSourceName` ("OTLP log ingestion") to be consistent

**Review by CelianR — APPROVED** [review]
LGTM for ci-experience files

**Review by FlorentClarret — APPROVED** [review]

File: comp/otelcol/otlp/components/exporter/datadogexporter/factory.go (hunk 5 above)
Comment (carlreid) [reply comp/otelcol/otlp/components/exporter/datadogexporter/factory.go]: Hey @songy23, is `otlp_log_ingestion` some internal Datadog source name that is handled/parsed as an integration?

**Pull request merged**